File: JAVA/2491. Divide Players Into Teams of Equal Skill.java
class Solution {
    public long dividePlayers(int[] s) {
        long t,a=0;
        int i=0,j=s.length-1;
        Arrays.sort(s);
        t=s[i]+s[j];
        while(i<j){
            if(s[i]+s[j]==t)
            a+=s[i++]*s[j--];
            else
            return -1;
        }
        return a;
    }
}

File: C++/2491. Divide Players Into Teams of Equal Skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& s) {
        long long a,d=0;
        sort(s.begin(),s.end());
        a=accumulate(s.begin(),s.end(),0);
        if(a%(s.size()/2)!=0)
        return -1;
        a=a/(s.size()/2);
        int i=0,j=s.size()-1;
        while(i<=j){
            if(s[i]+s[j]==a)
            d+=s[i++]*s[j--];
            else
            return -1;
        }
        return d;
    }
};